torch/csrc/autograd/anomaly_mode.h

```diff
@@ -22,6 +23,9 @@ struct TORCH_API AnomalyMetadata {
   virtual ~AnomalyMetadata();
   virtual void store_stack() = 0;
   virtual void print_stack(const std::string& current_node_name) = 0;
+  // using shared_ptr<void> instead of shared_ptr<Node> to avoid circular
+  // dependency between this file and "function.h"
+  virtual void assign_parent(const std::shared_ptr<void>& parent_node) = 0;
```

Comment:
Can't you just forward declare Node like we do in https://github.com/pytorch/pytorch/blob/73dcfc5e786258dd5beeff9c313d31213640a1de/torch/csrc/autograd/variable.h#L50 ?

Reply:
thanks! this is what I'm looking for